cabal-install/src/Distribution/Client/CmdStatus.hs

```diff
@@ -0,0 +1,381 @@
+import Distribution.Client.CmdErrorMessages
+import qualified Distribution.Client.InstallPlan as InstallPlan
+import Distribution.Client.NixStyleOptions
+         ( NixStyleFlags (..), nixStyleOptions, defaultNixStyleFlags )
+import Distribution.Client.ProjectOrchestration
+import Distribution.Client.ProjectPlanning
+import Distribution.Client.Setup
+         ( GlobalFlags, ConfigFlags(..), yesNoOpt )
+import Distribution.Client.Types
+         ( PackageSpecifier, PackageLocation )
+import Distribution.Client.TargetSelector
+         ( TargetSelectorProblem )
+import Distribution.Client.Utils.Json
+         ( (.=) )
+import qualified Distribution.Client.Utils.Json as Json
+import Distribution.Client.Version
+         ( cabalInstallVersion )
+
+import qualified Distribution.Compat.CharParsing as P
+import Distribution.ReadE
+         ( ReadE(ReadE), parsecToReadE )
+import Distribution.Simple.Command
+         ( CommandUI(..), option, reqArg, ShowOrParseArgs, OptionField )
+import Distribution.Simple.Compiler
+import Distribution.Simple.Program
+import Distribution.Simple.Flag
+         ( Flag(..), fromFlagOrDefault )
+import Distribution.Simple.Utils
+         ( wrapText, die', withOutputMarker, ordNub )
+import Distribution.Solver.Types.SourcePackage
+import Distribution.Types.UnitId
+import Distribution.Verbosity
+         ( normal )
+import Distribution.Version
+
+-------------------------------------------------------------------------------
+-- Command
+-------------------------------------------------------------------------------
+
+statusCommand :: CommandUI (NixStyleFlags StatusFlags)
+statusCommand = CommandUI
+  { commandName = "status"
+  , commandSynopsis = "Query for simple project information"
+  , commandDescription  = Just $ \_ -> wrapText $
+      "Query for available targets and project information such as project GHC."
+  , commandNotes = Just $ \pname ->
+        "Examples:\n"
+     ++ "  " ++ pname ++ " status --output-format=json --compiler\n"
+     ++ "    Print the compiler that is used for this project in the json format.\n"
+     ++ "  " ++ pname ++ " status --output-format=json --target=./src/Foo.hs\n"
+     ++ "    Print the unit-id of the component \"src/Foo.hs\" belongs to.\n"
+     ++ "  " ++ pname ++ " status --output-format=json --target=./src/Foo.hs\n"
+     ++ "    Print both, compiler information and unit-id for the given target.\n"
+     ++ "  " ++ pname ++ " status --output-format=json --target=./src/Foo.hs --target=./test/Bar.hs\n"
+     ++ "    Print unit-id location for multiple targets.\n"
+  , commandUsage = \pname ->
+      "Usage: " ++ pname ++ " status [FLAGS]\n"
+  , commandDefaultFlags = defaultNixStyleFlags defaultStatusFlags
+  , commandOptions      = nixStyleOptions statusOptions
+
+  }
+
+-------------------------------------------------------------------------------
+-- Flags
+-------------------------------------------------------------------------------
+
+data StatusOutputFormat
+  = JSON
```

Comment:
can we document since when a particular format is supported?

Reply:
Good addition!